### PR TITLE
fix failures on nightly 2

### PR DESCRIPTION
**Pull request by simeonschaub**
commits taken from #437 and #449, so this should probably not be squashed. Just fixes the tests and addresses some warnings on top of those PRs.

closes #437
closes #449

### CODE REVIEW

**Comment by timholy** [comment]
Since you've done this, let's do it this way! Thanks again for getting this going.

**Comment by simeonschaub** [comment]
I can also just rebase this after those two PRs are merged, if you'd prefer that

**Comment by timholy** [comment]
Just extra work. No need for that.

**Comment by timholy** [comment]
Is the email you made these commits with not registered with your GitHub account?

**Comment by simeonschaub** [comment]
> Is the email you made these commits with not registered with your GitHub account?

Yes, those were made on a different computer, and apparently, I haven't set up git properly yet there.

File: src/utils.jl [hunk 1]
```diff
@@ -248,12 +248,16 @@ end
 is_leaf(frame::Frame) = frame.callee === nothing
 
 function is_vararg_type(x)
-    if isa(x, Type)
-        (x <: Vararg && !(x <: Union{})) && return true
-        if isa(x, UnionAll)
-            x = Base.unwrap_unionall(x)
+    if isa(Vararg, Type)
```
Comment (aviatesk) [inline src/utils.jl:251]:
```suggestion
    @static if isa(Vararg, Type)
```

Comment (simeonschaub) [reply src/utils.jl:251]: Any functional reason to prefer `@static` here, or is it mostly stylistic?

Comment (aviatesk) [reply src/utils.jl]: yeah, it also may make it clearer that this branching because of version compatibility when we read this code later.

**Comment by simeonschaub** [comment]
Any clue what's going on with the Windows failure? Looks unrelated to this PR though.

**Comment by pfitzseb** [comment]
Looks like that `expanduser` related path issue that Tim tracked down, I think.

**Comment by timholy** [comment]
I'm good with this. Any objections if we merge? (Non-squash.)

**Comment by aviatesk** [comment]
I'm good too, I would appreciate it if we could have a new version that works on the current master soonish :p

**Comment by timholy** [comment]
For your Ctrl-F5 pleasure: https://github.com/JuliaRegistries/General/pull/26534

**Pull request merged**